publisher-sdk/src/main/java/com/criteo/publisher/csm/FileMetricRepository.java

```diff
@@ -104,22 +106,19 @@ boolean contains(@NonNull String impressionId) {
   /**
    * Atomically get or create a synchronized metric file on the given file.
    * <p>
-   * At most, one {@link SyncMetricFile} should exist for an underlying file. This method
-   * ensure this with a logic similar to {@link ConcurrentMap#computeIfAbsent(Object, Function)}.
-   * Note that this method is not directly used because it requires Android API level >= 24.
+   * At most, one {@link SyncMetricFile} should exist for an underlying file.
    *
    * @param metricFile underlying file to synchronized
    * @return unique instance of synchronized and atomic file over given one
    */
   @NonNull
   private SyncMetricFile getOrCreateMetricFile(@NonNull File metricFile) {
-    SyncMetricFile oldMetric = metricFileById.get(metricFile);
-    if (oldMetric == null) {
-      SyncMetricFile newMetric = directory.createSyncMetricFile(metricFile);
-      oldMetric = (oldMetric = metricFileById.putIfAbsent(metricFile, newMetric)) == null
-          ? newMetric : oldMetric;
-    }
-    return oldMetric;
+    return MapUtilKt.getOrCompute(metricFileById, metricFile, new Function0<SyncMetricFile>() {
```

Comment:
I would have preferred if we didn't include more changes on this commit than what is necessary. Can you separate the fix for the 3.10.2 and ship all the other changes later?

Reply:
Ok, I've put this in another commit, so the 3.10.2 could be made only with the two first commits:
- try catch in activity
- thread safe dependency provider